.github/workflows/stage.yaml

```diff
@@ -0,0 +1,25 @@
+name: Building and pushing to s3 stage
+
+on: 
+  pull_request:
+    # This relies on the default types [opened, synchronized, reopened]
+    branches: master
+  
+jobs:
+  build-dita:
+    name: build-dita
+    runs-on: ubuntu-latest
+    steps:
+    - name: Initialize repo
+      uses: actions/checkout@master
+    - name: Build HTML from source
+      run: .github/build_s3.sh
+    - name: Install s3-cli
+      uses: actions/aws/cli@master  
+    - name: Upload docs to the bucket
+      run: |
+        aws s3 sync html s3://puppet-docs-website-transfer/bolt-dev --delete
+        aws s3 ls --recursive s3://puppet-docs-website-transfer/bolt-dev
+      env:
+       AWS_ACCESS_KEY_ID: ${{ secrets.S3_ACCESS_KEY_ID }} 
```

Comment:
I don't know enough about GHA but are these really secret? What prevents someone from creating a PR that adds a step that sends these secrets somewhere?

I assume these creds are locked down enough that they only allow someone to push to an s3 bucket we don't really care about. However, IMO if they're not secret we shouldn't encrypt them someone may see that and put an important secret here too.

Reply:
The workflow that's executed will be whatever's on the primary (master) branch, not the one in the PR. The only way to test workflows is to run them in your own fork (which won't have that secret) before PRing back.

Reply:
Now if you have a workflow that executes code from the PR - like building and running that code - then could it read the secret and print it? Maybe. Although I *think* GitHub actions actually filters for the secret value and covers it in asterisks.